cmd/servewallet/main.go

```diff
@@ -199,6 +199,10 @@ func main() {
 	fmt.Printf("Listening on: http://localhost:%d\n", port)
 
 	if *useSimulator {
+		// The simulator is only allowed when running in testnet mode.
+		if *mainnet {
```

Comment:
Probably the same problem as https://github.com/BitBoxSwiss/bitbox-wallet-app/pull/3621#discussion_r2435972289?

Reply:
Thanks, I think in this case we can simply use `backend.Testing()` instead, since the `backend` is already constructed and `Testing()` takes care of the tesnet config option